test/TestAssets/AttachmentProcessorDataCollector/TestExtensionTypesAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using AttachmentProcessorDataCollector;
+using Microsoft.VisualStudio.TestPlatform;
+
+[assembly: TestExtensionTypes(typeof(SampleDataCollectorV1))]
+[assembly: TestExtensionTypesV2(ExtensionInfo.ExtensionType, ExtensionInfo.ExtensionIdentifier, typeof(SampleDataCollectorV1), 1, "futureUnused")]
+[assembly: TestExtensionTypesV2(ExtensionInfo.ExtensionType, ExtensionInfo.ExtensionIdentifier, typeof(SampleDataCollectorV2), 2)]
+
+namespace Microsoft.VisualStudio.TestPlatform
+{
+    using System;
+
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false, AllowMultiple = false)]
+    internal sealed class TestExtensionTypesAttribute : Attribute
+    {
+        public TestExtensionTypesAttribute(params Type[] types)
+        {
+            this.Types = types;
+        }
+
+        public Type[] Types { get; }
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false, AllowMultiple = true)]
+    internal sealed class TestExtensionTypesV2Attribute : Attribute
+    {
+        public string ExtensionType { get; }
```

Comment:
>ExtensionType [](http://example.com/codeflow?start=22&length=13)

I'm assuming this attribute would make it into the product after a review. Would be nice to add comments and example formats of what each of these signify.

Reply:
The idea is to "not ship" the attribute inside the object model. 
We will ship the "documentation" of it very similar to what we're doing inside coverlet for code coverage exclusion https://github.com/coverlet-coverage/coverlet/blob/master/src/coverlet.core/Instrumentation/Instrumenter.cs#L66

In this way we won't risk to ship a version of attribute and in future change the signature and come back to the same "load issue" we have today.